Tasks/Common/azure-arm-rest/azure-arm-mysql.ts

```diff
@@ -0,0 +1,320 @@
+     * @param firewallRuleName      firewall rule name to be deleted 
+     * @param callback              response callback 
+     */
+    public get(resourceGroupName: string, serverName: string, firewallRuleName: string, callback?:  azureServiceClient.ApiCallback) {
+        var client = this.client;
+        if (!callback) {
+            throw new Error(tl.loc("CallbackCannotBeNull"));
+        }
+
+        // Validate
+        try {
+            this.client.isValidResourceGroupName(resourceGroupName);
+            if(!this.isNameValid(serverName)){
+                throw new Error(tl.loc("MysqlServerNameCannotBeEmpty"));
+            }
+            if(!this.isNameValid(firewallRuleName)){
+                throw new Error(tl.loc("FirewallRuleNameCannotBeNull"));
+            }
+        }
+        catch (error) {
+            return callback(error);
+        }
+
+        var httpRequest = new webClient.WebRequest();
+        httpRequest.method = 'GET';
+        httpRequest.headers = {};
+        httpRequest.uri = this.client.getRequestUri(
+            '//subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/servers/{serverName}/firewallRules/{firewallRuleName}',
+            {
+                '{resourceGroupName}': resourceGroupName,
+                '{serverName}': serverName,
+                '{firewallRuleName}': firewallRuleName
+            });
+        tl.debug("Calling get firewall ");
+        this.client.beginRequest(httpRequest).then(async (response) => {
+            var deferred = Q.defer<azureServiceClient.ApiResult>();
+            var statusCode = response.statusCode;
+            tl.debug("Response for get firewall rule " +statusCode);
+            if (statusCode === 200) {
+                // Generate Response
+                tl.debug("Response for get firewall rule " + JSON.stringify(response));
+                deferred.resolve(new azureServiceClient.ApiResult(null, response));
+            }
+            else {
+                // Generate exception
+                deferred.reject(new azureServiceClient.ApiResult(azureServiceClient.ToError(response)));
+            }
+            return deferred.promise;
+        }).then((apiResult: azureServiceClient.ApiResult) => callback(apiResult.error, apiResult.result),
+            (error) => callback(error));
+    }
+
+
+    /**
+     * Retry get call to check firewall rule has added or not
+     * @param resourceGroupName     resource group name of mysql server 
+     * @param serverName            mysql server name
+     * @param firewallRuleName      firewall rule name to be deleted 
+     * @param retryOption           no of time to retry
+     */
+    private _recursiveGetCall(resourceGroupName: string, serverName: string, firewallRuleName: string, retryOption: number) : Q.Promise<azureServiceClient.ApiResult>{
+        var deferred = Q.defer<azureServiceClient.ApiResult>();
+        setTimeout(() => {
```

Comment:
also, can we rename the function?
Suggestion to have a while loop rather than a recursive call here to avoid complexity.

Reply:
If we use while loop then main thread which will not blocked on network call as it is async call so it will again come for 2nd iteration and then for 3rd iteration . Then it will be similar to making three call in parallel.